### ISSUE TITLE
Local testing of temporal concatenation (5 iterations)

### ISSUE THREAD

**Issue by jeremymanning** [ContextLab/giblet-responses#27]
# Local Testing Before Cluster Deployment

## Objective

Verify temporal concatenation implementation works correctly by training locally for 5 iterations.

## Test Configuration

**Small subset for quick validation:**
- 1 subject (e.g., subject 1)
- 20 TRs (30 seconds)
- 5 training iterations
- Batch size: 4
- CPU or single GPU

### Data Loading
- [ ] Video features load without errors
- [ ] Audio features load without errors
- [ ] Text features load without errors
- [ ] fMRI features load without errors
- [ ] All dimensions consistent

### Model Forward Pass
- [ ] Encoder processes all modalities
- [ ] Bottleneck has correct shape (batch, 2048)
- [ ] Decoder reconstructs all modalities
- [ ] Output shapes match input shapes

### Training Loop
- [ ] Loss calculation works
- [ ] Gradients flow correctly
- [ ] No NaN/Inf values
- [ ] Parameters update
- [ ] 5 iterations complete successfully

### Checkpointing
- [ ] Checkpoint saves correctly
- [ ] Checkpoint loads correctly
- [ ] Training can resume from checkpoint

### Memory/Performance
- [ ] Peak memory usage reasonable
- [ ] No memory leaks
- [ ] Iteration time acceptable

## Success Criteria

**ALL must pass:**
1. ✅ 5 iterations complete without errors
2. ✅ Loss decreases (even slightly)
3. ✅ No dimension mismatches
4. ✅ Checkpoint save/load works
5. ✅ Memory usage acceptable (<8GB for local)

**Only after all pass:** Deploy to cluster

## Test Script

Create `test_local_training.py`:
```python
# Load small subset
dataset = MultimodalDataset(
    'data',
    subjects=[1],
    max_trs=20
)

# Create model
model = MultimodalAutoencoder()

# Train for 5 iterations
for i in range(5):
    loss = train_step(...)
    print(f"Iteration {i+1}: Loss = {loss:.4f}")

# Save checkpoint
torch.save(model.state_dict(), 'test_checkpoint.pt')

# Verify reload
model.load_state_dict(torch.load('test_checkpoint.pt'))
```

## Timeline

- Setup test: 30 min
- Run test: 30 min
- Debug if needed: 1-2 hours
- Verify: 30 min

**Total:** 2-4 hours

## Deliverables

1. test_local_training.py script
2. Test results log
3. Confirmation all checks pass
4. Ready to deploy to cluster

Refs #26

**Comment by jeremymanning** [comment]
Deferred - Will test during actual training run.

Local 5-iteration test can be done in future if needed for debugging.

Current approach: Test on cluster directly with monitoring.

Ready to launch training with Batch 1 temporal concatenation fixes.
